esp/services/ws_decoupledlogging/ws_decoupledloggingservice.cpp

```diff
@@ -102,38 +102,13 @@ bool CWSDecoupledLogEx::onGetLogAgentSetting(IEspContext& context, IEspGetLogAge
 {
     try
     {
-        IArrayOf<IEspLogAgentGroupSetting> groupSettingResp;
-        IArrayOf<IConstLogAgentGroup>& groups = req.getGroups();
-        if (!groups.ordinality())
-        {
-            for (auto ml : logGroups)
-                getSettingsForLoggingAgentsInGroup(ml.second, nullptr, groupSettingResp);
-        }
-        else
-        {
-            ForEachItemIn(i, groups)
-            {
-                IConstLogAgentGroup& g = groups.item(i);
-                const char* gName = g.getGroupName();
-                if (isEmptyString(gName))
-                    throw MakeStringException(ECLWATCH_INVALID_INPUT, "Group name not specified.");
-
-                auto match = logGroups.find(gName);
-                if (match != logGroups.end())
-                {
-                    StringArray& agentNames = g.getAgentNames();
-                    getSettingsForLoggingAgentsInGroup(match->second, &agentNames, groupSettingResp);
-                }
-                else
-                {
-                    Owned<IEspLogAgentGroupSetting> groupSetting = createLogAgentGroupSetting();
-                    groupSetting->setGroupName(gName);
-                    groupSetting->setGroupStatus("NotFound");
-                    groupSettingResp.append(*groupSetting.getClear());
-                }
-            }
-        }
-        resp.setSettings(groupSettingResp);
+        LogAgentAction action;
+        action.type = CLogAgentActions_GetSettings;
+
+        CLogAgentActionResults results;
+        WSDecoupledLogGetSettings act(action, results);
+        act.doAction(logGroups, req.getGroups());
```

Comment:
I will move the try-catch into doAction().

Reply:
If a throw is added before or after the doAction() in future, we may move the try-catch outside.

Reply:
I would rather see doAction call doActionSelf (or something like that), where the common code like exception handling is shared in "doAction" and the "Self" method does the class-specific logic. Personally, I don't like seeing a lot of redundancy because there was a special case.